src/util/renderers/renderer.builder.ts

```diff
@@ -0,0 +1,48 @@
+import MediaWiki from './../../MediaWiki.js'
+import { Renderer } from './abstract.renderer.js'
+import { VisualEditorRenderer } from './visual-editor.renderer.js'
+import { WikimediaDesktopRenderer } from './wikimedia-desktop.renderer.js'
+import { RendererBuilderOptions } from './../saveArticles.js'
+
+export class RendererBuilder {
+  private renderApi: 'VisualEditor' | 'WikimediaDesktop' | 'WikimediaMobile'
+  private renderMode: 'auto' | 'desktop' | 'mobile' | 'specific'
+  private mw: MediaWiki
+
+  public createRenderer(options: RendererBuilderOptions): Renderer {
+    const { mw, RendererMode, RendererAPI } = options
+
+    this.mw = mw
+    this.renderMode = RendererMode
+    this.renderApi = RendererAPI
+
+    switch (this.renderMode) {
+      case 'desktop':
+        return new WikimediaDesktopRenderer()
```

Comment:
VisualEditor output is Desktop. we should return WikimediaDesktopRenderer() if possible, otherwise VisualEditor (if possible). We should generate an exception (and handle it) if somehow no appropriate end-point is available.

Reply:
In that case, there is 'auto' render mode. Or do you expect that the ``desktop` mode will also check capabilities?

Reply:
The renderer builder has been updated along with tests.